components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +67,527 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self.isLayoutDirty = YES;
   } else {
-    _image = nil;
-    [self.iconImageViewContainer removeFromSuperview];
-    self.iconImageViewContainer = nil;
+    self.iconImageView.image = icon;
   }
   [self setNeedsLayout];
 }
 
-- (void)setNumberOfButtons:(NSUInteger)numberOfButtons {
-  NSAssert(numberOfButtons <= kNumberOfButtonsLimit,
-           @"%@ class doesn't support more than %lu buttons", NSStringFromClass([self class]),
-           (unsigned long)kNumberOfButtonsLimit);
-  if (numberOfButtons != self.buttons.count) {
-    NSMutableArray *mutableButtons = [self.buttons mutableCopy];
-    if (numberOfButtons > self.buttons.count) {
-      for (NSUInteger index = self.buttons.count; index < numberOfButtons; ++index) {
-        MDCButton *button = [[MDCButton alloc] init];
-        [mutableButtons addObject:button];
-        [self addSubview:button];
-      }
-    } else if (numberOfButtons < self.buttons.count) {
-      for (NSUInteger index = numberOfButtons; index < self.buttons.count; ++index) {
-        MDCButton *lastObject = [mutableButtons lastObject];
-        [lastObject removeFromSuperview];
-        [mutableButtons removeLastObject];
-      }
-    }
-    self.buttons = [mutableButtons copy];
+- (UIImage *)icon {
+  return self.iconImageView.image;
+}
+
+- (void)setButtons:(NSMutableArray<MDCButton *> *_Nonnull)buttons {
+  NSAssert(buttons.count <= kNumberOfButtonsLimit, @"%@ doesn't support more than %lu buttons",
+           NSStringFromClass([self class]), (unsigned long)kNumberOfButtonsLimit);
+  _buttons = buttons;
+  for (MDCButton *button in buttons) {
+    button.translatesAutoresizingMaskIntoConstraints = NO;
+    [self.containerView addSubview:button];
   }
+  self.isLayoutDirty = YES;
+}
+
+- (void)setBackgroundColor:(UIColor *)backgroundColor {
+  [super setBackgroundColor:backgroundColor];
+  self.containerView.backgroundColor = backgroundColor;
+}
+
+- (void)setTextFont:(UIFont *)textFont {
+  self.textLabel.font = textFont;
+}
+
+- (UIFont *)textFont {
+  return self.textLabel.font;
+}
+
+- (void)setTextColor:(UIColor *)textColor {
+  self.textLabel.textColor = textColor;
+}
+
+- (UIColor *)textColor {
+  return self.textLabel.textColor;
 }
 
-- (NSUInteger)numberOfButtons {
-  return self.buttons.count;
+- (void)setIconTintColor:(UIColor *)iconTintColor {
+  self.iconImageView.tintColor = iconTintColor;
+}
+
+- (UIColor *)iconTintColor {
+  return self.iconImageView.tintColor;
 }
 
 #pragma mark - UIView overrides
 
 - (CGSize)sizeThatFits:(CGSize)size {
-  self.layout = [[MDCBannerViewLayout alloc] initWithSizeToFit:size
-                                                     textLabel:self.textLabel
-                                                imageContainer:self.iconImageViewContainer
-                                                       buttons:self.buttons];
-  return [self.layout frameSize];
+  return CGSizeMake([UIScreen mainScreen].bounds.size.width, self.layout.size.height);
```

Comment:
Should this be using `preferredContentWidth` instead of the screen width? I notice this PR is removing a lot of UIScreen inspection, but not here

Reply:
It was intended to be made screen width. The banner should stay as wide as screen width, but the content width is defined by user by `preferredContentWidth`. The space between `preferredContentWidth` and screen edge can be considered as white margin.